### DISCUSSION TITLE
Representing phisical units together with the data

### DISCUSSION THREAD

**Discussion by steinnymir** [OpenCOMPES/sed#37] [Ideas]
An useful feature would be to keep track of units of the physical quantities represented.
A python package designed to handle these is [Pint](https://pint.readthedocs.io/en/stable/). It basically allows to assign units to numbers and numpy arrays.

An interesting project to hook up to is [pint-xarray](https://pint-xarray.readthedocs.io/en/latest/). This can handle the xarray objects we are using extensively, adding physical units to them.

I open this here as a discussion to test how much this could be of interest, and if it is worth investing the time in developing this.

**Comment by rettigl** [comment]
This needs to be done eventually. The Nexus framework we are building in Fairmat has proper units handling and tracking as prerequisite, so gathering and housekeeping the units also in this package does not seem like a bad idea, on the other hand we should avoid doing too much twice.
